docs/hsi.xml

```diff
@@ -0,0 +1,875 @@
+                Intel CSME Security Review Cumulative Update
+              </ulink>
+            </listitem>
+          </itemizedlist>
+        </para>
+      </note>
+    </refsect3>
+
+
+    <refsect3 id="org.fwupd.hsi.IntelDci">
+      <title>Intel DCI</title>
+      <para>
+        Newer Intel CPUs support debugging over USB3 via a proprietary Direct
+        Connection Interface (DCI) with the use of off-the-shelf hardware.
+        DCI should always be disabled and locked on production hardware.
+      </para>
+      <itemizedlist>
+        <listitem>
+          <para>
+            For HSI-1 this should be disabled. <emphasis>v1.5.0</emphasis>
+          </para>
+        </listitem>
+        <listitem>
+          <para>
+            For HSI-2 this should be locked. <emphasis>v1.5.0</emphasis>
+          </para>
+        </listitem>
+      </itemizedlist>
+      <note>
+        <para>
+          See also:
+          <itemizedlist>
+            <listitem>
+              <ulink url="https://www.intel.co.uk/content/www/uk/en/support/articles/000029393/processors.html">
+                Intel Direct Connect Interface
+              </ulink>
+            </listitem>
+            <listitem>
+              <ulink url="https://github.com/chipsec/chipsec/blob/master/chipsec/cfg/8086/pch_4xxlp.xml#L270">
+                Chipsec 4xxlp register definitions
+              </ulink>
+            </listitem>
+            <listitem>
+              <ulink url="https://github.com/riscv/riscv-edk2-platforms/blob/85a50de1b459d1d6644a402081120770aa6dd8c7/Silicon/Intel/CoffeelakeSiliconPkg/Pch/Include/Register/PchRegsDci.h">
+                RISC-V EDK PCH register definitions
+              </ulink>
+            </listitem>
+          </itemizedlist>
+        </para>
+      </note>
+    </refsect3>
+
+    <refsect3 id="org.fwupd.hsi.Tpm.ReconstructionPcr0">
+      <title>PCR0 TPM Event Log Reconstruction</title>
+      <para>
+        The TPM event log records which events are registered for the PCR0 hash.
+        When reconstructed the event log values should always match the TPM PCR0.
+        If extra events are included in the event log, or some are missing,
+        the reconstitution will fail.
+      </para>
+      <itemizedlist>
+        <listitem>
+          <para>
+            For HSI-2 this should match the TPM-provided PCR0 <emphasis>v1.5.0</emphasis>
+          </para>
+        </listitem>
+      </itemizedlist>
+      <note>
+        <para>
+          See also:
+          <itemizedlist>
+            <listitem>
+              <ulink url="https://www.kernel.org/doc/html/latest/security/tpm/tpm_event_log.html">
+                Linux Kernel TPM Documentation
+              </ulink>
+            </listitem>
+          </itemizedlist>
+        </para>
+      </note>
+    </refsect3>
+    <refsect3 id="org.fwupd.hsi.AcpiDmar">
+      <title>Kernel DMA protection</title>
```

Comment:
This is pre-boot specifically, no?

Reply:
I think this is a good example of where the spec and fwupd were disagreeing on what stuff actually meant; and one of the big reasons I wanted to bring it all into one place (the other, being able to add URLs to the various HSI attrs to link back to the spec...).

This is of course pre-boot IOMMU. How about this:

      <title>Pre-boot DMA protection</title>
      <para>
        The IOMMU on modern systems is used to mitigate against DMA attacks.
        All I/O for devices capable of DMA is mapped into a private virtual
        memory region.
        The ACPI DMAR table is used to set up pre-boot DMA protection which
        eliminates some firmware attacks.
      </para>

Reply:
👍